Pcap++/src/PcapRemoteDeviceList.cpp

```diff
@@ -84,8 +84,9 @@ PcapRemoteDeviceList* PcapRemoteDeviceList::getRemoteDeviceList(const IPAddress&
 
 	for (pcap_if_t* currInterface = interfaceList.get(); currInterface != nullptr; currInterface = currInterface->next)
 	{
-		std::unique_ptr<PcapRemoteDevice> pNewRemoteDevice = std::unique_ptr<PcapRemoteDevice>(
-			new PcapRemoteDevice(currInterface, resultList->m_RemoteAuthentication, resultList->getRemoteMachineIpAddress(), resultList->getRemoteMachinePort()));
+		auto pNewRemoteDevice = std::unique_ptr<PcapRemoteDevice>(
```

Comment:
I understand why you put the `new PcapRemoteDevice` inside `std::uique_ptr`, but I think it's safe to do the following here, which should be easier to read(?)
```
auto newRemoteDevice = new PcapRemoteDevice(currInterface, resultList->m_RemoteAuthentication, resultList->getRemoteMachineIpAddress(), resultList->getRemoteMachinePort()));
std::unique_ptr<PcapRemoteDevice> pNewRemoteDevic(newRemoteDevice);
```
maybe the first line can be split by the arguments as well.

Reply:
Hmm, It is possible but I kinda dislike doubling up on the pointer variables.

I suppose we can do this, if it is more clear?:
```
auto pNewRemoteDevice = std::unique_ptr<PcapRemoteDevice>(
	new PcapRemoteDevice(
		currInterface,
		resultList->m_RemoteAuthentication, 
		resultList->getRemoteMachineIpAddress(), 
		resultList->getRemoteMachinePort()
	)
);
```
But it is just going to get squashed to this:
```
auto pNewRemoteDevice = std::unique_ptr<PcapRemoteDevice>(
    new PcapRemoteDevice(currInterface, resultList->m_RemoteAuthentication,
                         resultList->getRemoteMachineIpAddress(), resultList->getRemoteMachinePort()));
```
when the formatting comes to it.

Reply:
I think it's clearer. don't hesitate to use `// clang-format off` if the formatted is no good.

Reply:
Fixed the formatting in 6a4c330